### ISSUE TITLE
Add QC feature for `sct_deepseg`

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#4436]
As more and more people are going to use the contrast-agnostic model (among other models), we should probably start adding the QC feature

**Comment by joshuacwnewton** [comment]
I'm presuming that, because the contrast agnostic update is a high priority for the pre-ISMRM release (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4440), we also want to make sure that the QC is done, too? (Hence the high priority label?)

I'm happy to self-assign for now, but if @mguaypaq is planning to take look at this as part of the ongoing QC refactor, I can step back and focus on reviewing his work instead. :)

**Comment by mguaypaq** [comment]
@joshuacwnewton go for it! And since we already merged the initial `qc2.py`, feel free to use either the old structure or the new, we can always port it later if need be.

**Comment by jcohenadad** [comment]
> I'm presuming that, because the contrast agnostic update is a high priority for the pre-ISMRM release (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4440), we also want to make sure that the QC is done, too? (Hence the high priority label?)

Yes, but to be honest this is more of an egoistic 'high priority' label. I've been using the contrast-agnostic model in other projects, and every time I have to add an [extra line of code](https://github.com/spinal-cord-7t/coil-qc-code/blob/e6b338f8d89b546228e2d5228055e429d16d42cb/data_processing.ipynb#L116) it hurts 😅  (not mentioning that the QC indicates 'sct_deepseg_sc', which is both wrong and confusing, when prototyping analysis pipelines and comparing with the _actual_ sct_deepseg_sc)
